### PR TITLE
fix: Single app info form 

### PR DESCRIPTION
**Pull request by sulha199**
## Summary of change
- utilizing DOM attributes, Mutation Observer & InterractionObserver
- when user click the resubmit, it will find the first `appInfoForm`, and then update `display-form` DOM attribute
- Mutation Observer that listen to the attribute changes
- refactor render method and break it into 2 methods
- workaround on details element

## Related issues
- https://github.com/supertokens/main-website/issues/879
- #408 

## Checklist
- [ ] Algolia search needs to be updated? (If there is a new sub docs project, then yes)
- [ ] Sitemap needs to be updated? (If there is a new sub docs project, then yes)
- [x] Checked for broken links? (Run `cd v2 && MODE=production npx docusaurus build`)
- [ ] Changes required to the demo apps corresponding to the docs?

## Remaining TODOs for this PR
NA

### CODE REVIEW

**Comment by sulha199** [comment]
Below is the file that I use for the test case

```mdx
import FrontendSDKTabs from "/src/components/tabs/FrontendSDKTabs";
import TabItem from "@theme/TabItem";
import AppInfoForm from "/src/components/appInfoForm";
import NpmOrScriptTabs from "/src/components/tabs/NpmOrScriptTabs"

# Test Case on AppInfoForm

<dialog><AppInfoForm askForAPIDomain>0th AppInfoForm inside dialog</AppInfoForm></dialog>

<details>
  <summary>There is form inside</summary>
  1st AppInfoForm inside detail
  <AppInfoForm askForAPIDomain>Content of 1st AppInfoForm inside detail</AppInfoForm>
</details>
2nd AppInfoForm on outer
<AppInfoForm askForAPIDomain>Content of 2nd AppInfoForm on outer</AppInfoForm>

<FrontendSDKTabs>
  <TabItem value="reactjs">
    <details>
      <summary>There is form inside</summary>
      3rd AppInfoForm inside detail that's inside a tab 
      <AppInfoForm askForAPIDomain>Content of 3rd AppInfoForm inside detail that's inside a tab </AppInfoForm>
    </details>
    4th AppInfoForm's that's inside a tab
    <AppInfoForm askForAPIDomain>Content of 4th AppInfoForm's that's inside a tab</AppInfoForm>

  </TabItem>
  <TabItem value="vanillajs">
    <AppInfoForm askForAPIDomain>
      
      AppInfoForm's content
    </AppInfoForm>
  </TabItem>
  <TabItem value="vue">

    <NpmOrScriptTabs>
      <TabItem value="npm">

        <AppInfoForm askForAPIDomain>AppInfoForm's content</AppInfoForm>

        ```bash
        npm i -s supertokens-web-js
        ```

      </TabItem>
      <TabItem value="script">

      ```bash
      <script src="https://cdn.jsdelivr.net/gh/supertokens/supertokens-web-js/bundle/website.js"></script>
      <script src="https://cdn.jsdelivr.net/gh/supertokens/supertokens-web-js/bundle/supertokens.js"></script>
      <script src="https://cdn.jsdelivr.net/gh/supertokens/supertokens-web-js/bundle/session.js"></script>
      ```

      </TabItem>
    </NpmOrScriptTabs>
  </TabItem>
  <TabItem value="react-native">
    <AppInfoForm askForAPIDomain>AppInfoForm's content</AppInfoForm>
  </TabItem>

  <TabItem value="angular">
    <AppInfoForm askForAPIDomain>
      AppInfoForm's content
    </AppInfoForm>

  </TabItem>
</FrontendSDKTabs>

```

File: v2/src/components/appInfoForm/index.tsx [hunk 1]
```diff
@@ -724,4 +785,26 @@ export default class AppInfoForm extends React.PureComponent<PropsWithChildren<P
...
+                top: window.scrollY + (this.containerRef.current?.parentElement?.getBoundingClientRect().top ?? 0),
+                behavior: 'smooth'
+            });
+        }
+    }
+
+    private onReceiveAttribute(mutations: MutationRecord[]) {
+        if (isReceivingFirstFormAttr(mutations)) {
+            this.setState({ firstAppInfoForm: this.isFirstVisibleAppInfoForm() })
+        }
```
Comment (sulha199) [inline v2/src/components/appInfoForm/index.tsx:801]: the `if (isReceivingFirstFormAttr(mutations))` is the fix for case of the following structure
```tsx
<details>
  <summary>There is form inside</summary>  
  <AppInfoForm askForAPIDomain>Content of 1st AppInfoForm inside detail</AppInfoForm>
</details>

<AppInfoForm askForAPIDomain>Content of 2nd AppInfoForm on outer</AppInfoForm>
```

File: v2/src/components/appInfoForm/index.tsx [hunk 2]
```diff
@@ -33,14 +33,47 @@ type State = {
     netlifyApiRouteUsed: boolean,
     showWebsiteBasePath: boolean,
     showAPIBasePath: boolean
+    firstAppInfoForm?: boolean
     // TODO: Add more fields here
 };
 
+type ResubmitParams = {
```
Comment (nkshah2) [inline v2/src/components/appInfoForm/index.tsx:40]: Add comments above the changes you made to explain what they do

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: v2/src/components/appInfoForm/index.tsx [hunk 3]
```diff
@@ -33,14 +33,61 @@ type State = {
     netlifyApiRouteUsed: boolean,
```
Comment (nkshah2) [inline v2/src/components/appInfoForm/index.tsx:33]: The first line of the form is not true anymore, we should remove it. In this case the styling from the first line should apple to the second line

Screenshot 2022-08-31 at 11 33 35 AM

File: v2/src/components/appInfoForm/index.tsx [hunk 4]
```diff
@@ -33,14 +33,61 @@ type State = {
...
     showWebsiteBasePath: boolean,
     showAPIBasePath: boolean
```
Comment (nkshah2) [inline v2/src/components/appInfoForm/index.tsx:35]: When both the form and code snippets are visible, the bottom spacing for the form is incorrect. It should use the same spacing as the resubmit state

Screenshot 2022-08-31 at 10 21 36 AM

Screenshot 2022-08-31 at 10 22 33 AM

File: v2/src/components/appInfoForm/index.tsx [hunk 5]
```diff
@@ -33,14 +33,61 @@ type State = {
...
+    /** Whether current element is the first visible AppInfoForm */
+    firstAppInfoForm?: boolean
     // TODO: Add more fields here
```
Comment (nkshah2) [inline v2/src/components/appInfoForm/index.tsx:38]: When clicking resubmit the user should be taken to the first visible instance of the form, but in this example the user is scrolled to a section that does not have the app info form (even though technically it does, it isnt visible and should not be considered the first instance on screen)

https://user-images.githubusercontent.com/18233774/187606064-5910b0c3-ca41-415b-b8de-a324453e8d7c.mp4

Comment (nkshah2) [reply v2/src/components/appInfoForm/index.tsx:38]: To reproduce:

1. Navigate to frontend quick setup
2. In step 2) Fill the form for the React tab
3. Scroll to step 5)
4. Click resubmit

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: v2/src/components/appInfoForm/index.tsx (hunk 2 above)
Comment (sulha199) [reply v2/src/components/appInfoForm/index.tsx:41]: added

File: v2/src/components/appInfoForm/index.tsx (hunk 3 above)
Comment (sulha199) [reply v2/src/components/appInfoForm/index.tsx:33]: updated

File: v2/src/components/appInfoForm/index.tsx (hunk 4 above)
Comment (sulha199) [reply v2/src/components/appInfoForm/index.tsx:35]: updated

File: v2/src/components/appInfoForm/index.tsx (hunk 5 above)
Comment (sulha199) [reply v2/src/components/appInfoForm/index.tsx:38]: fixed

File: v2/src/components/appInfoForm/index.tsx (hunk 2 above)
Comment (nkshah2) [reply v2/src/components/appInfoForm/index.tsx:41]: Add more context to your comments

File: v2/src/components/appInfoForm/index.tsx [hunk 6]
```diff
@@ -33,14 +33,59 @@ type State = {
     netlifyApiRouteUsed: boolean,
     showWebsiteBasePath: boolean,
     showAPIBasePath: boolean
+    /** Whether current element is the first visible AppInfoForm */
```
Comment (nkshah2) [inline v2/src/components/appInfoForm/index.tsx:36]: Screenshot 2022-09-05 at 10 22 21 AM

Screenshot 2022-09-05 at 10 22 28 AM

The form does not save all the inputs to storage

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: v2/src/components/appInfoForm/index.tsx (hunk 2 above)
Comment (sulha199) [reply v2/src/components/appInfoForm/index.tsx:58]: its added

File: v2/src/components/appInfoForm/index.tsx (hunk 6 above)
Comment (sulha199) [reply v2/src/components/appInfoForm/index.tsx:53]: its fixed

**Comment by nkshah2** [comment]
@sulha199 Some general feedback from others

- For placeholders add an underscore before and after to make it look like a placeholder
- For all instances except for the first form dont show resubmit state
- Ideally we should see if it is possible to change the color of the text for the placeholder values only

**Comment by nkshah2** [comment]
Closing in favor of https://github.com/supertokens/docs/pull/457

**Pull request closed without merging**